### ISSUE TITLE
Overwriting pretrained model file

### ISSUE THREAD

**Issue by mangleddata** [ludwig-ai/ludwig#1075]
This is probably a feature request that is related to https://github.com/ludwig-ai/ludwig/issues/1028. I believe overwriting all the model parameters when pretrained model is loaded could be improved. Here's the scenario.
Run 1: Run 100 epochs, set early_stop to be say 10 epochs. Model file is ready now.
Run 2: New data and new model file, but the only change in the model is setting early_stop to be 2 epochs say. Current implementation would simply override the training parameters and can keep running for 100 epoch or early stop: 10 is reached.

Allowing few parameters such as number of epochs or early stop values to be modified will save time in subsequent runs. I am right now modifying the model_hyperparameters.json file from run 1 to accomplish the same effect. When slow learning rate is used, each epoch can take a lot of time - so earlier stop values on subsequent runs could be helpful.

**Comment by jimthompson5802** [comment]
@mangleddata  Thank you for the comment.
